src/main/java/com/careerfair/q/service/queue/implementation/QueueServiceImpl.java

```diff
@@ -157,6 +172,23 @@ public GetQueueStatusResponse getQueueStatus(String studentId) {
 
     @Override
     public AddQueueResponse addQueue(String companyId, String employeeId, Role role) {
+        try {
+            com.careerfair.q.model.db.Employee employee = employeeFirebase.getEmployeeWithId(
+                    employeeId);
+
+            if (!companyId.equals(employee.getCompanyId())) {
+                throw new InvalidRequestException("Employee with employee id=" + employeeId +
+                        " is not associated with company with company id=" + companyId);
+            } else if (employee.getRole() != role) {
+                throw new InvalidRequestException("Employee with employee id=" + employeeId +
+                        " is not associated with the role=" + role);
+            }
+
+            fairFirebase.getCompanyWithId("1tgJU9lL0kdzrElX7C3Y", companyId);
```

Comment:
Add a constant - with a comment saying that this doesn't mean it works for multiple concurrent career fairs.